### DISCUSSION TITLE
QuickAdd++ -like functionality: Prompt for input for template values

### DISCUSSION THREAD

**Discussion by vcavallo** [epwalsh/obsidian.nvim#653] [Ideas]
There are a few powerful Obsidian plugins for templates, with Templater and QuickAdd++ being among the best. One of their greatest features is getting a prompt for entering values for template fields.

Given a template like:

```
I went to ...
{{ place }}
with ...
{{ people }}
for 
{{ reason }}
```

A powerful template plugin would, upon creating a new note with this template, prompt sequentially for values for `place`, `people` and `reason` and then fill those values in.

Is something similar with a combination of vim command line and "completion"-style popups?

I know we can define custom template values based on functions, but these are limited to being somewhat "predefined" (even if dynamic, like a date function) and don't allow live entry of arbitrary values.

I'm not suggesting that obsidian.vim recognize the dynamic values of a template and prompt for them - that seems way outside scope. But is it possible with LUA and vimscript to define "prompty" functions in the obsidian.vim template configs?

**Comment by epwalsh** [comment]
Hey @vcavallo,

> I'm not suggesting that obsidian.vim recognize the dynamic values of a template and prompt for them

Actually I really like that idea and it wouldn't be too hard to implement. When you get a chance could you turn this into a feature request issue so it doesn't fall off my radar?

**Reply by vcavallo** [reply]
done! https://github.com/epwalsh/obsidian.nvim/issues/654
